utils/fixture.js

```diff
@@ -3,21 +3,13 @@ const isEmpty = require('lodash.isempty')
 
 function setFixture (mock, fixture, name) {
   fixture = fixture && fixture[name] ? fixture[name] : undefined
-  if (typeof fixture === 'undefined') {
+
+  if (fixture === undefined) {
     return mock
   }
 
   if (Array.isArray(fixture)) {
-    const fixtureArr = []
-    fixture.forEach(val => {
-      if (mock[0] === null) {
-        return val
-      }
-
-      const result = handleObjectFixture(mock[0], val)
-      fixtureArr.push(Object.assign({}, result))
-    })
-    return fixtureArr
+    return fixture.map(val => isObject(val) ? Object.assign({}, handleObjectFixture(mock[0], val)) : val)
   } else {
```

Comment:
What do you think if we change this block, for this?
```js
  if (Array.isArray(fixture)) {
    return fixture.map(val => {
      if (isObject(val)) {
        return Object.assign({}, handleObjectFixture(mock[0], val))
      }

      return scalarValidation(val, fixture, name)
    })
  } else if (isObject(fixture)) {
    return handleObjectFixture(mock, fixture)
  } else {
    return scalarValidation(mock, fixture, name)
  }
```
and,
```js
function scalarValidation(mock, fixture, name) {
  if (Array.isArray(mock) && !Array.isArray(fixture)) {
    throw new Error(`${name} fixture is not an array and it should be one.`)
  }

  if (Array.isArray(fixture)) {
    fixture.forEach(val => {
      if (typeof mock !== typeof val) {
        throw new Error(`${name} fixture is not the same type as the document.`)
      }
    })
  } else if (typeof mock !== typeof fixture) {
    throw new Error(`${name} fixture is not the same type as the document.`)
  }
  
  return mock
}
```
That way we can mock `scalars`, try adding a fixture on [this](https://github.com/EasyGraphQL/easygraphql-tester/pull/69/files#diff-91ecf4f8217f439ff0dcf2f68fa16f67R816) test

Reply:
@estrada9166  Refactoring this block using your code snippet. Still, I had some new tests not passing, when involving arrays or scalars.

Here un update handling arrays (a bit better) and scalars. Including some "edge" cases like:
- `{data: null, errors: [...]}`
- `{data: {getUsers: 'invalid'}}`
- `{data: {getUsers: ['invalid'}}`
- `{data: {getInt: 99}}`


```js
function setFixture (mock, fixture, name) {
  if (fixture == null) {
    return null
  }

  fixture = fixture && fixture[name] ? fixture[name] : undefined

  if (fixture === undefined) {
    return mock
  }

  if (Array.isArray(fixture)) {
    return fixture.map((val, index) => {
      if (isObject(val)) {
        return Object.assign({}, handleObjectFixture(mock[index], val))
      }

      return scalarValidation(mock[index], val, name)
    })
  } else if (isObject(fixture)) {
    return handleObjectFixture(mock, fixture)
  } else {
    return scalarValidation(mock, fixture, name)
  }
}

function scalarValidation(mock, fixture, name) {
  if (Array.isArray(mock) && !Array.isArray(fixture)) {
    throw new Error(`${name} fixture is not an array and it should be one.`)
  }

  if (Array.isArray(fixture)) {
    fixture.forEach((val, index) => {
      if (typeof mock[index] !== typeof val) {
        throw new Error(`${name} fixture is not the same type as the document.xxx`)
      }
    })
  } else if (typeof mock !== typeof fixture) {
    throw new Error(`${name} fixture is not the same type as the document.`)
  }

  return fixture !== undefined ? fixture : mock
}
```

Reply:
Sorry, my bad... we can remove this on `scalarValidation`:
```js
 if (Array.isArray(mock) && !Array.isArray(fixture)) {
    throw new Error(`${name} fixture is not an array and it should be one.`)
  }

  if (Array.isArray(fixture)) {
    fixture.forEach((val, index) => {
      if (typeof mock[index] !== typeof val) {
        throw new Error(`${name} fixture is not the same type as the document.`)
      }
    })
  } 
```
[Here](https://github.com/facebook/graphql/issues/423) is the main reason!

Some, other minor changes there :)
```js
fixture = fixture && fixture[name] ? fixture[name] : undefined

if (fixture === null) {
    return null
}
```
It should be after the fixture because, it sets the fixture of the requested operation, so it'll access that info... and change `==` for `===` so we can be sure that data is equal!!

Also, #70 this can be useful!

Reply:
The order is check first for null ... because else can not handle `{data: null, errors: [....]}` else will return the mock.

```
js
if (fixture === null) {
    return null
}

fixture = fixture && fixture[name] ? fixture[name] : undefined
```

Reply:
Sorry, I miss that... that's true, it's when the `{data: null, errors: [....]}`